core/server/fluxruntime.go

```diff
@@ -93,6 +94,39 @@ func (cs *coreServer) ListFluxRuntimeObjects(ctx context.Context, msg *pb.ListFl
 	return &pb.ListFluxRuntimeObjectsResponse{Deployments: results, Errors: respErrors}, nil
 }
 
+func (cs *coreServer) ListFluxCrds(ctx context.Context, msg *pb.ListFluxCrdsRequest) (*pb.ListFluxCrdsResponse, error) {
+	clustersClient, err := cs.clientsFactory.GetImpersonatedClient(ctx, auth.Principal(ctx))
+	if err != nil {
+		return nil, fmt.Errorf("error getting impersonating client: %w", err)
+	}
+
+	var results []*pb.Crd
+
+	respErrors := []*pb.ListError{}
+
+	opts := client.MatchingLabels{
+		coretypes.PartOfLabel: FluxNamespacePartOf,
+	}
+
+	list := &apiextensions.CustomResourceDefinitionList{}
+
+	if err := clustersClient.List(ctx, msg.ClusterName, list, opts); err != nil {
+		respErrors = append(respErrors, &pb.ListError{ClusterName: msg.ClusterName, Message: fmt.Sprintf("%s, %s", ErrListingDeployments.Error(), err)})
+	}
+
+	for _, d := range list.Items {
+		r := &pb.Crd{
+			Name:        d.Spec.Names.Plural + "." + d.Spec.Group,
+			Version:     d.Spec.Versions[0].Name,
```

Comment:
I worry `versions` might be empty. We should make sure we have tests for that.

Have a look at https://github.com/weaveworks/weave-gitops/blob/main/core/server/fluxruntime_test.go#L84 - it creates a few objects and plugs them into a `fake` client, creates a new core server using the fake client, and then calls the endpoint in the client.